cli/queryflags/flags.go

```diff
@@ -36,11 +37,13 @@ func (f *Flags) ParseSourcesAndInputs(paths []string) ([]string, ast.Seq, bool,
 			// and appears to start with a from or yield operator.
 			// Otherwise, consider it a file.
 			if query, err := compiler.Parse(src, f.Includes...); err == nil {
-				if isFrom(query) {
-					return nil, query, false, nil
-				}
-				if isYield(query) {
-					return nil, query, true, nil
+				if a, err := semantic.NewAnalyzer(context.Background(), query, nil, nil); err == nil {
+					if a.HasSource() {
+						return nil, query, false, nil
+					}
+					if a.StartsWithYield() {
+						return nil, query, true, nil
+					}
 				}
```

Comment:
This interface feels off because the methods interrogate the DAG, not the semantic analyzer.

How about something like this instead?
```suggestion
				if queryDag, err := semantic.Analyze(context.Background(), query, nil, nil); err == nil {
					if semantic.HasSource(queryDag) {
						return nil, query, false, nil
					}
					if semantic.StartsWithYield(queryDag) {
						return nil, query, true, nil
					}
				}
```

Reply:
I can dig that.

Reply:
ok